client/src/utilities/chart.js

```diff
@@ -1,6 +1,11 @@
+import Immutable from 'immutable';
+
 export function getChartData(visualisation, datasets) {
   const { datasetId, spec } = visualisation;
-  const dataset = datasets[datasetId];
```

Comment:
I think the dataset should always be an Immutable object. Is this not always the case?